File: My neural network project #2/layer.py
import random
import math
from neuron import Neuron
# this is the class for the layers

class Layer(object):
        ##### Code that runs #####
        def __init__(self, numNeu):
            self.neurons = []
            for i in range(numNeu): #attach neurons to layer
                self.neurons.append(Neuron())

        def resetRunAndChange(self):
            for i in range(len(self.neurons)):
                self.neurons[i].ran = False
                self.neurons[i].changed = False

        def hasANeuChanged(self):
            result = False;
            #check all neurons to see if they have fired
            for i in range(len(self.neurons)):
                if(self.neurons[i].changed == True):
                    result = True

            return result

        def haveAllNeuRun(self):
            result = True
            for i in range(len(self.neurons)):
                if (self.neurons[i].ran == False):
                    result = False

            return result

File: My neural network project #2/Web App/layer.js

//this is the class for the layers

function Layer(numNeu){

        this.neurons = [];
        for (var i = 0; i < numNeu; i++){ //attach neurons to layer
            this.neurons.push(new Neuron());
        }

        this.resetRunAndChange = function(){
            for (var i = 0; i < this.neurons.length; i++){
                this.neurons[i].ran = false;
                this.neurons[i].changed = false;
            }
        }

        this.hasANeuChanged = function(){
            var result = false;
            //check all neurons to see if they have fired
            for (var i = 0; i < this.neurons.length; i++){
                if(this.neurons[i].changed == true){
                    result = true;
                }
            }

            return result;
        }

        this.haveAllNeuRun = function(){
            var result = true;
            for (var i = 0; i < this.neurons.length; i++){
                if (this.neurons[i].ran == false){
                    result = false;
                }
            }

            return result;
        }
}